in_fake_critical_section.hpp

```diff
@@ -9,6 +9,48 @@
 
 namespace unodb {
 
+// fake version of read_critical_section used to align unodb::db and
+// unodb::olc_db code.
+class [[nodiscard]] fake_read_critical_section final {
```

Comment:
Adding this class asks to rename the file. `concurrency_stubs.hpp`? Not sure

Reply:
There are similar patterns with optimistic_lock.hpp.  I would not change things around right now.

Reply:
OK, I'll note to do this after merging